src/lib/tree/data-source/flat-data-source.ts

```diff
@@ -50,7 +50,7 @@ export class MatTreeFlattener<T, F> {
   constructor(public transformFunction: (node: T, level: number) => F,
               public getLevel: (node: F) => number,
               public isExpandable: (node: F) => boolean,
-              public getChildren: (node: T) => Observable<T[]> | T[]) {}
+              public getChildren: (node: T) => Observable<T[]> | T[] | undefined) {}
```

Comment:
The fix for this file should really be its own PR since it's a separate bug fix from cleaning up the examples. It also needs a unit test. I'd allow `null` here as well.

Reply:
Would it make sense to allow `ReadonlyArray<T>` here as well? We had a fix a while ago that added it to all the data source usages.

Reply:
That would be good. The fix would be something like "relax allowed types for tree traversal function"